packages/mdc-select/component.ts

```diff
@@ -376,68 +348,40 @@ export class MDCSelect extends MDCComponent<MDCSelectFoundation> implements MDCR
     // tslint:disable:object-literal-sort-keys Methods should be in the same order as the adapter interface.
     const adapter: MDCRippleAdapter = {
       ...MDCRipple.createAdapter(this),
-      registerInteractionHandler: (evtType, handler) => this.targetElement_.addEventListener(evtType, handler),
-      deregisterInteractionHandler: (evtType, handler) => this.targetElement_.removeEventListener(evtType, handler),
+      registerInteractionHandler: (evtType, handler) => this.selectedText_.addEventListener(evtType, handler),
+      deregisterInteractionHandler: (evtType, handler) => this.selectedText_.removeEventListener(evtType, handler),
     };
     // tslint:enable:object-literal-sort-keys
     return new MDCRipple(this.root_, new MDCRippleFoundation(adapter));
   }
 
-  private getNativeSelectAdapterMethods_() {
-    // tslint:disable:object-literal-sort-keys Methods should be in the same order as the adapter interface.
-    return {
-      getValue: () => this.nativeControl_!.value,
-      setValue: (value: string) => {
-        this.nativeControl_!.value = value;
-      },
-      openMenu: () => undefined,
-      closeMenu: () => undefined,
-      isMenuOpen: () => false,
-      setSelectedIndex: (index: number) => {
-        this.nativeControl_!.selectedIndex = index;
-      },
-      setDisabled: (isDisabled: boolean) => {
-        this.nativeControl_!.disabled = isDisabled;
-      },
-      setValid: (isValid: boolean) => {
-        if (isValid) {
-          this.root_.classList.remove(cssClasses.INVALID);
-        } else {
-          this.root_.classList.add(cssClasses.INVALID);
-        }
-      },
-      checkValidity: () => this.nativeControl_!.checkValidity(),
-    };
-    // tslint:enable:object-literal-sort-keys
-  }
-
   private getEnhancedSelectAdapterMethods_() {
     // tslint:disable:object-literal-sort-keys Methods should be in the same order as the adapter interface.
     return {
       getValue: () => {
-        const listItem = this.menuElement_!.querySelector(strings.SELECTED_ITEM_SELECTOR);
+        const listItem = this.menuElement_.querySelector(strings.SELECTED_ITEM_SELECTOR);
         if (listItem && listItem.hasAttribute(strings.ENHANCED_VALUE_ATTR)) {
           return listItem.getAttribute(strings.ENHANCED_VALUE_ATTR) || '';
         }
         return '';
       },
       setValue: (value: string) => {
-        const element = this.menuElement_!.querySelector(`[${strings.ENHANCED_VALUE_ATTR}="${value}"]`);
-        this.setEnhancedSelectedIndex_(element ? this.menu_!.items.indexOf(element) : -1);
+        const element = this.menuElement_.querySelector(`[${strings.ENHANCED_VALUE_ATTR}="${value}"]`);
+        this.setEnhancedSelectedIndex_(element ? this.menu_.items.indexOf(element) : -1);
       },
       openMenu: () => {
-        if (this.menu_ && !this.menu_.open) {
+        if (!this.menu_.open) {
```

Comment:
We might not need this check here. We can refactor this in upcoming PRs.

Reply:
Noted.